components/org.wso2.identity.event.websubhub.publisher/src/main/java/org/wso2/identity/event/websubhub/publisher/service/WebSubTopicManagerImpl.java

```diff
@@ -128,24 +131,82 @@ private void makeTopicMgtAPICall(String topic, String webSubHubBaseUrl, String o
-                    try {
-                        Thread.sleep(RETRY_DELAY_MS);
-                    } catch (InterruptedException ie) {
-                        Thread.currentThread().interrupt();
-                        throw handleServerException(ERROR_REGISTERING_HUB_TOPIC, ie, topic, tenantDomain);
-                    }
+                    log.debug("Retrying topic management API call, attempt " + attempt + " for topic: " + topic);
+                    Thread.sleep(RETRY_DELAY_MS);
                     continue;
                 }
-                handleTopicMgtResponse(response, httpPost, topic, operation, requestStartTime);
-                break; // Success or handled error
-            } catch (IOException | WebSubAdapterException e) {
+
+                if (responseCode == HttpStatus.SC_OK) {
+                    HttpEntity entity = response.getEntity();
+                    WebSubHubCorrelationLogUtils.triggerCorrelationLogForResponse(httpPost, requestStartTime,
+                            WebSubHubCorrelationLogUtils.RequestStatus.COMPLETED.getStatus(),
+                            String.valueOf(responseCode), responsePhrase);
+                    if (entity != null) {
+                        String responseString = EntityUtils.toString(entity, StandardCharsets.UTF_8);
+                        if (RESPONSE_FOR_SUCCESSFUL_OPERATION.equals(responseString)) {
+                            log.debug("Success WebSub Hub operation: " + operation + ", topic: " + topic);
+                        } else {
+                            throw handleServerException(ERROR_INVALID_RESPONSE_FROM_WEBSUB_HUB, null, topic,
+                                    operation, responseString);
+                        }
+                    } else {
+                        String message =
+                                String.format(ERROR_EMPTY_RESPONSE_FROM_WEBSUB_HUB.getDescription(), topic, operation);
+                        throw new WebSubAdapterServerException(message, ERROR_EMPTY_RESPONSE_FROM_WEBSUB_HUB.getCode());
+                    }
+                } else if ((responseCode == HttpStatus.SC_CONFLICT && operation.equals(REGISTER)) ||
+                        (responseCode == HttpStatus.SC_NOT_FOUND && operation.equals(DEREGISTER))) {
+                    HttpEntity entity = response.getEntity();
+                    String responseString = "";
+                    WebSubHubCorrelationLogUtils.triggerCorrelationLogForResponse(httpPost, requestStartTime,
+                            WebSubHubCorrelationLogUtils.RequestStatus.FAILED.getStatus(),
+                            String.valueOf(responseCode), responsePhrase);
+                    if (entity != null) {
+                        responseString = EntityUtils.toString(entity, StandardCharsets.UTF_8);
+                    }
+                    log.warn(String.format(ERROR_INVALID_RESPONSE_FROM_WEBSUB_HUB.getDescription(),
```

Comment:
don't add a warn log. This should be a debug and you can add a diagnostic later

Reply:
fixed with https://github.com/wso2-extensions/identity-event-publishers/pull/25/commits/67d32091145f93afc04a6815cd9e2ef1828a00c3